### PR TITLE
Changing the data and resource files to be generated instead of hand crafted, allows adding new blocks with much less effort

### PR DESCRIPTION
**Pull request by campbebj**
Changing the data and resource files to be generated instead of hand crafted, allows adding new blocks with much less effort

New blocks added to the type file can have their data and resources generated with the gradle "runData" command before you run "build".

rerequest of https://github.com/DinnerBeef/Compressium/pull/25 with bugfix as requested of DinnerBeef

### CODE REVIEW

**Comment by DinnerBeef** [comment]
Ok few things.
1. Please do not add any blocks to what I all ready have. I plan on keeping this more of the core block in the game that you would tend to have more of than anything else. 
2. I know did this for 1.16 but I would rather have this in the 1.17 build before instead.
3. There is a 1.17 Branch ATM that mostly works only bug ATM is some textures with the item and not the blocks.

**Comment by DinnerBeef** [comment]
Also one more thing you can not push all the Data files in your PR so I can look at the changes easier I can run that before I push

**Comment by campbebj** [comment]
Sure. ill make a branch that has just the core blocks. 

Ill see about doing teh 1.17 as well. might take a bit for me to refactor for 1.17 changes (some mods were more affected than others)

How do you want me to handle the data files then. make the PR only have the java code at first so you can review, and then add teh data files in later for push , since you wanted to have the data files committed?

**Comment by DinnerBeef** [comment]
Just dont push the data files I will just pull an make a new push with them. Since there might be a few things i want to change anyway. As I said thought before is is all ready i 1.17 branch so it might just be easier to copy an paste your code an change it in that branch. As long as you can find the renamed stuff :)

**Comment by DinnerBeef** [comment]
O one more thing I did remove the Honey block for 1.17 version. PS i do plan on adding more blocks just not in this mod

**Comment by campbebj** [comment]
makes sense. ill do the 1.17 stuff and post a non generated change. 


Why a seperate mod for more blocks? curious. part of the reason i wanted to do this change was because i could add more blocks really easily and just recompile to get all the new files (i got tired of hand crafting them after 3 files)

**Comment by DinnerBeef** [comment]
Not everyone needs all the blocks and since this mod adds lots of blocks already, I don't want people to have 1000s of block the don't need. Most likely going to have 1-2 "special" ones then have one more mod that adds the rest of the blocks that those don't add.
TLDR: People have options on which blocks they want

PS. I wrote something that makes them for me outside of this mod since I did not know how to Data-Gens
________________________________

**Comment by campbebj** [comment]
Makes sense that people would like different mods. Have you thought about making it so the block registration is dynamic based on a config? i did reasearch and foudn that block generation cant be dynamic, but i imagine block registration might be able to be dynamic. Thoughts? if you're interested i could pursue that as a follow up.

**Comment by DinnerBeef** [comment]
I get that idea of this, however. Not everyone that is going to play with this mod will know how to do it and I want to make sure that people can add more blocks to the pack if they want without having to learn how to do configuration magic.
So while that could be cool
Mods = Easier For People to add to a pack that they did not make
________________________________

**Comment by campbebj** [comment]
so your thinking you would rather Mod A be the core blocks. and Mod B be everything else, which if you dont know how config files would is definitly the simplest answer. even though some mod menus exist for this sort of thing (though that doesnt help with servers)

I guess im thinking in terms of powers users, 
For someone like me to say "oh i want more blocks, i cant help but build too many over powered farms"
or
for people who have a modpack and want to say "i want this modpack to have compressed blocks for these specific blocks", or even further to say "oh i only want compression level 7 as the mox not 9"

**Comment by DinnerBeef** [comment]
I am trying to keep this mod as sample as it can be. So yea just a different mods for different blocks is what I want. Though if you want you can add Max Compression level to a config "for all blocks in the mods" with a max level of 9 since i fell like that is a good size to stop at. Default at 9 though

**Comment by campbebj** [comment]
Are you thinking the new mod would be "depends on compressium" ?

And yeah 9 is a good max i dont disagree. ill see if that makes sense ot add.

**Comment by DinnerBeef** [comment]
no they wont all mods would be there own. Again for users that just want some block they dont need all of them

**Comment by campbebj** [comment]
Would it make sense then to build a compressium library that can be build IN to the other mods rather than as a dependency, so that you dont have duplicate code in each mod?

**Comment by DinnerBeef** [comment]
Yea it's not that much code I think it's fine

Get Outlook for Android<https://aka.ms/AAb9ysg>

________________________________

**Comment by DinnerBeef** [comment]
See #29
